Automattic/jetpack
# Actions: Add changelog entry check for Repo Gardening action

#### Changes proposed in this Pull Request:

* Add check for changelog entries
* Refactor checks and recommendations rendering
* Fix few issues with Milestone detection and outptut.

#### Jetpack product discussion


n/a
#### Does this pull request change what data or activity we track or use?


n/a
#### Testing instructions:






* Check the bot comment in this test PR: https://github.com/Automattic/jetpack/pull/19200
* Make changes to projects with and without changelogger, make sure the check reflects that
